api4/post_test.go

```diff
@@ -3229,3 +3229,205 @@ func TestPostReminder(t *testing.T) {
 
 	require.Truef(t, caught, "User should have received %s event", model.WebsocketEventEphemeralMessage)
 }
+
+func TestMovePost(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	client := th.Client
+	userWSClient, err := th.CreateWebSocketClient()
+	require.NoError(t, err)
+	defer userWSClient.Close()
+	userWSClient.Listen()
+
+	t.Run("Users should receive post deleted event for the root post being moved", func(t *testing.T) {
+		resp, err := client.MoveThread(th.BasicPost.Id, &model.MoveThreadParams{
+			ChannelId: th.BasicChannel2.Id,
+		})
+
+		require.NoError(t, err)
+		CheckOKStatus(t, resp)
+
+		var caught bool
+		func() {
+			for {
+				select {
```

Comment:
nit: I think the `func()` and `for` blocks are unnecessary: we want this to be blocking, so we can just `select` directly on the two channels. That would also let us avoid a `caught` variable altogether, and just fail the test directly in the `time.After` block.